### ISSUE TITLE
`<wa-dialog>` and `<wa-drawer>` use unlabeled ARIA dialog elements

### ISSUE THREAD

**Issue by kasper-i** [shoelace-style/webawesome#2785]
### Describe the bug
The `<wa-dialog>` and `<wa-drawer>` components both use the ARIA `dialog` role, but are lacking an ARIA name. According to the Firefox Accessibility inspector a label should be provided. Can we use the `label` property as the ARIA name?

### Screenshots

<img width="1344" height="1033" alt="Image" src="https://github.com/user-attachments/assets/42f20058-31b7-4c83-8412-d301e8d75a97" />
